File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.java
import java.util.*;

class Solution {
    public int solution(int n, int[][] costs) {
        HashSet<Integer> set = new HashSet<>();
        int answer = 0;

        Arrays.sort(costs, Comparator.comparingInt(o -> o[2]));
        set.add(costs[0][0]);

        while (set.size() != n) {
            for (int[] cost : costs) {
                if ((set.contains(cost[0]) && !set.contains(cost[1])) || (!set.contains(cost[0]) && set.contains(cost[1]))) {
                    set.add(cost[0]);
                    set.add(cost[1]);
                    answer += cost[2];
                    break;
                }
            }
        }

        return answer;
    }
}

File: 프로그래머스/3/42861. 섬 연결하기/섬 연결하기.kt
class Solution {
    fun solution(n: Int, costs: Array<IntArray>): Int {
        val arr = costs.sortedBy { it[2] }
        val set = mutableSetOf(arr[0][0])
        var cost = 0

        while (set.size != n) {
            run breaking@ {
                arr.forEach {
                    if ((set.contains(it[0]) && !set.contains(it[1])) || (!set.contains(it[0]) && set.contains(it[1]))) {
                        cost += it[2]
                        set.addAll(it.dropLast(1))
                        return@breaking
                    }
                }
            }
        }

        return cost
    }
}